source/common/access_log/access_log_formatter.cc

```diff
@@ -68,7 +68,7 @@ std::string FormatterImpl::format(const Http::HeaderMap& request_headers,
     log_line += provider->format(request_headers, response_headers, response_trailers, stream_info);
   }
 
-  return log_line;
+  return absl::StrCat(log_line, "\n");
```

Comment:
I think you need to condition this on whether it already ends in a newline. Also, it'll be more efficient to do that when constructing the formatter. See AccessLogFormatParser::parse(). I'd just add a PlainStringFormatter to the end of the vector that contains a `\n`, but only do that if it does not already end in `\n`

Reply:
Excellent! I was hoping that I was just missing something obvious here.